numpy/ma/core.py

```diff
@@ -1613,7 +1613,11 @@ def make_mask(m, copy=False, shrink=True, dtype=MaskType):
             else:
                 result = m
         else:
-            result = np.array(m, dtype=dtype, copy=copy)
+            # existence of fields implies True
+            if m.dtype.fields and np.dtype(dtype) == np.bool_:
+                result = np.ones(m.shape, dtype=dtype)
```

Comment:
I couldn't help myself and ended up doing some debugging...

I think there is still a problem with MA masks after #8590, specifically with the `make_mask` method. This seems wrong:
```python
>>> a = np.zeros(4, dtype='i4,f4')
>>> np.ma.make_mask(a)
array([ True,  True,  True,  True], dtype=bool)
```
Shouldn't it return 
```python
array([(False, False), (False, False), (False, False), (False, False)], 
      dtype=[('f0', '?'), ('f1', '?')])
```
?

Actually, `make_mask` did that before #8590 too.